### PR TITLE
Remove Cake Build

### PR DESCRIPTION
**Pull request by Nirmal4G**
### Changes

We no longer require Cake build nor we use it extensively!
Update Azure Pipelines (CI YAML) Script to be more modern and runnable across platforms.
Replaced the headers check with a PowerShell script mimicking the Cake script's `UpdateHeaders` task functionality.

### Notes

**Squash/Rebase merge** if possible.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -18,66 +18,64 @@ jobs:
...
+      versionSpec: 6.0.0
 
   # Install NerdBank GitVersioning
   - task: DotNetCoreCLI@2
     displayName: Install NBGV tool
     inputs:
       command: custom
       custom: tool
-      arguments: install --tool-path . nbgv
+      arguments: install -g nbgv
```
Comment (Sergio0694) [inline azure-pipelines.yml:31]: Are there any differences between using the two different formats here?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -18,66 +18,64 @@ jobs:
...
+
+  # Set Build Version
   - script: nbgv cloud
     displayName: Set NBGV version
 
   # Verify headers
-  - powershell: .\build\build.ps1 -Target Verify
+  - pwsh: build/Update-Headers.ps1 -Verify
```
Comment (Sergio0694) [inline azure-pipelines.yml:38]: Why the change from `powershell` to `pwsh` here?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -18,66 +18,64 @@ jobs:
...
     displayName: Verify headers
 
   # Build solution
-  - powershell: dotnet build -c Release
+  - script: dotnet build -c Release
```
Comment (Sergio0694) [inline azure-pipelines.yml:42]: Why the change from `powershell` to `script` here?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -18,66 +18,64 @@ jobs:
...
     condition: always()
 
   # Create the NuGet package(s)
-  - powershell: dotnet pack --configuration Release
+  - script: dotnet pack -c Release
     displayName: Create NuGet package(s)
 
   # Sign package(s)
-  - task: PowerShell@2
+  - pwsh: build/Sign-Package.ps1
```
Comment (Sergio0694) [inline azure-pipelines.yml:70]: Why did you remove the `task` here?

File: azure-pipelines.yml [hunk 5]
```diff
@@ -18,66 +18,64 @@ jobs:
...
       SignClientUser: $(SignClientUser)
       SignClientSecret: $(SignClientSecret)
-      ArtifactDirectory: bin\nupkg
-    condition: and(succeeded(), not(eq(variables['build.reason'], 'PullRequest')), not(eq(variables['SignClientSecret'], '')), not(eq(variables['SignClientUser'], '')))
+      ArtifactDirectory: bin/nupkg
+    condition: and(succeeded(), not(eq(variables['Build.Reason'], 'PullRequest')), not(eq(variables['SignClientSecret'], '')), not(eq(variables['SignClientUser'], '')))
 
   # Publish build artifacts
-  - task: PublishPipelineArtifact@1
+  - publish: bin/nupkg
```
Comment (Sergio0694) [inline azure-pipelines.yml:79]: Same as above, why was `task` removed here?

**Review by Sergio0694 — COMMENTED** [review]
A few questions, but overall I'm really happy about this change, thanks! 😄

File: azure-pipelines.yml (hunk 1 above)
Comment (Nirmal4G) [reply azure-pipelines.yml:31]: It's a global install. keeps the repo folder clean! I also read it helps in caching!!

File: azure-pipelines.yml (hunk 2 above)
Comment (Nirmal4G) [reply azure-pipelines.yml:38]: The script I wrote uses a few features that are only available in PowerShell Core (_v6+_). But feel free to update the script to work with Windows PowerShell (_v5.1_). And then you can use `powershell` task!

File: azure-pipelines.yml (hunk 4 above)
Comment (Nirmal4G) [reply azure-pipelines.yml:70]: shorthand

File: azure-pipelines.yml (hunk 5 above)
Comment (Nirmal4G) [reply azure-pipelines.yml:79]: same, shorthand

File: azure-pipelines.yml (hunk 3 above)
Comment (Nirmal4G) [reply azure-pipelines.yml:42]: `pwsh` runs in `script` but `powershell` doesn't. Just avoiding pipeline overhead. We can use `pwsh` but ehh... it's simple, predictable and it works, at least for simple invocations.

**Review by Sergio0694 — APPROVED** [review]
LGTM, thanks! :shipit:
